int32[] neuron_ids
float64 ts